### PR TITLE
Add ability to specify the name of an existing Placement or PlacementRule

### LINKED ISSUE #50: Reusing PlacementRule files across multiple Generators fails in ArgoCD
Attempting to use ArgoCD to manage ACM Policies.  Running kustomize where multiple generators set placementRulePath to the same file results in this error.

> rpc error: code = Unknown desc = Manifest generation error (cached): `kustomize build .acm-configuration/operators --enable-alpha-plugins` failed exit status 1: Error: accumulating resources: accumulation err='accumulating resources from 'dynatrace': '.acm-configuration/operators/dynatrace' must resolve to a file': recursed merging from path '.acm-configuration/operators/dynatrace': may not add resource with an already registered id: apps.open-cluster-management.io_v1_PlacementRule|smbc-acm-policies|placement-dev-apps-cluster

Steps to reproduce:
1) Setup two Policy Generators that deploy to the same namespace and both point to the same file for placementRulePath.
2) Run kustomize build command against kustomization.yaml which includes both resources from step 1

Reproducer available https://github.com/brian-jarvis/test-placement

### PR DESCRIPTION
**Pull request by brian-jarvis**
This allows managing all Placements/PlacementRules in the ACM hub separate from the generated policies that use them.

Fixes #50 

Example usage available at https://github.com/brian-jarvis/test-placement/tree/reuse-placement

Signed-off-by: Brian Jarvis <bjarvis@redhat.com>

### CODE REVIEW

File: docs/policygenerator-reference.yaml
Comment (dhaiducek) [inline docs/policygenerator-reference.yaml:225]: This can be removed once this PR is reviewed/merged:
- https://github.com/stolostron/policy-generator-plugin/pull/52

File: internal/plugin.go
Comment (dhaiducek) [inline internal/plugin.go:615]:
```suggestion
			"policyDefaults must specify only one of placement selector, placement path, or placement name",
```

File: internal/plugin.go
Comment (dhaiducek) [inline internal/plugin.go:752]:
```suggestion
				"policy %s must specify only one of placement selector, placement path, or placement name", policy.Name,
```

File: internal/plugin.go
Comment (dhaiducek) [inline internal/plugin.go:829]:
```suggestion
				"policySet %s must specify only one of placement selector, placement path, or placement name", plcset.Name,
```

File: docs/policygenerator-reference.yaml
Comment (brian-jarvis) [reply docs/policygenerator-reference.yaml:225]: Thank you, I will merge that change in once it is reviewed/merged

File: docs/policygenerator-reference.yaml
Comment (mprahl) [inline docs/policygenerator-reference.yaml:217]:
```suggestion
      # namespace as the PolicySet will be generated for.  It is the responsibility of the administrator to 
```

File: docs/policygenerator-reference.yaml
Comment (mprahl) [inline docs/policygenerator-reference.yaml:221]:
```suggestion
      # Advanced configuration to use a placement rule that already exists in the cluster in the same
```

File: docs/policygenerator-reference.yaml
Comment (mprahl) [inline docs/policygenerator-reference.yaml:222]:
```suggestion
      # namespace as the PolicySet will be generated for.  It is the responsibility of the administrator to 
```

File: docs/policygenerator-reference.yaml
Comment (mprahl) [inline docs/policygenerator-reference.yaml:72]: Could you also document these under `policies[0].placement`?

File: internal/plugin.go
Comment (mprahl) [inline internal/plugin.go:483]:
```suggestion
		// If both cluster label selectors and placement path/name aren't set, then use the defaults with a
```

File: internal/plugin.go
Comment (mprahl) [inline internal/plugin.go:484]:
```suggestion
		// priority on placement path followed by placement name.
```

File: internal/plugin.go
Comment (mprahl) [inline internal/plugin.go:493]:
```suggestion
			// Else if both cluster selectors and placement rule path/name aren't set, then use the defaults with a
```

File: internal/plugin.go
Comment (mprahl) [inline internal/plugin.go:494]:
```suggestion
			// priority on placement rule path followed by placement rule name.
```

File: internal/plugin.go
Comment (mprahl) [inline internal/plugin.go:619]: This should also account for when the placement path and placement name are both set.

File: internal/plugin_test.go
Comment (mprahl) [inline internal/plugin_test.go:189]: The patch is irrelevant for this test. Could you please remove it?

**Review by mprahl — CHANGES_REQUESTED** [review]
Thanks for the contribution! Just a few minor changes to make.

File: internal/plugin_test.go
Comment (brian-jarvis) [reply internal/plugin_test.go]: Removed patch

File: internal/plugin.go
Comment (mprahl) [inline internal/plugin.go:622]: I think this would be easier to read:
```suggestion
	defaultPlacementOptions := 0
	if len(p.PolicyDefaults.Placement.LabelSelector) != 0 || len(p.PolicyDefaults.Placement.ClusterSelectors) != 0 {
		defaultPlacementOptions += 1
	}
	if p.PolicyDefaults.Placement.PlacementRulePath != "" || p.PolicyDefaults.Placement.PlacementPath != "" {
		defaultPlacementOptions += 1
	}
	if p.PolicyDefaults.Placement.PlacementRuleName != "" || p.PolicyDefaults.Placement.PlacementName != "" {
		defaultPlacementOptions += 1
	}

	if defaultPlacementOptions > 1 {
```

File: internal/plugin.go
Comment (mprahl) [inline internal/plugin.go:840]: Same as above.

**Comment by mprahl** [comment]
@brian-jarvis could you also please squash all of your commits to a single commit message of `Add ability to specify the name of an existing Placement or PlacementRule`.

**Comment by mprahl** [comment]
@brian-jarvis could you please fix the linting error?

**Comment by brian-jarvis** [comment]
> @brian-jarvis could you please fix the linting error?

Fixed.

**Review by mprahl — APPROVED** [review]
Thanks for the contribution!

/hold for @dhaiducek to also take a look.

**Review by dhaiducek — APPROVED** [review]
LGTM! Thanks for these updates, @brian-jarvis!

**Pull request merged**
